backend/src/services/release.ts

```diff
@@ -295,6 +331,133 @@ export async function getReleaseBySemver(user: UserInterface, modelId: string, s
   return release
 }
 
+function getQuerySyntax(querySemver: string | undefined, modelID: string) {
+  //NOTE if we are adding advanced query for range, we ignore metadata as they aren't currently supported by our range query functionality
+  if (querySemver === undefined) {
+    return {
+      modelId: modelID,
+    }
+  }
+
+  const semverRangeStandardised = semver.validRange(querySemver, { includePrerelease: false })
+
+  //1. test if valid
+  //2. if so, split
+  //3. check for inclusivity
+  //4. create query
+
+  //1
+  if (!semverRangeStandardised) {
+    throw BadReq(`Semver range, '${querySemver}' is invalid `)
+  }
+
+  //2
+  const [lowerSemver, upperSemver] = semverRangeStandardised.split(' ')
+
+  //3
+  //inclusive
+  const lowerSemverTrimmed = lowerSemver.replace('>=', '')
+  let upperSemverTrimmed, inclusivity
+  if (upperSemver) {
+    if (upperSemver.includes('=')) {
+      //remove and extract
+      upperSemverTrimmed = upperSemver.replace('<=', '')
+      inclusivity = true
+    } else {
+      //exclusive
+      //remain
+      upperSemverTrimmed = upperSemver.replace('<', '')
+      inclusivity = false
+    }
+  }
+
+  //4
+  const lowerSemverObj = semverStringToObject(lowerSemverTrimmed)
+  let upperSemverObj
+  if (upperSemverTrimmed) {
+    upperSemverObj = semverStringToObject(upperSemverTrimmed)
+  }
+
+  if (inclusivity)
```

Comment:
This function is a lot more simpler and readable, good job!
To improve upon this a bit more I would move all the logic needed to convert the `semverRangeStandardised` content into a form that can be used to generate this mongo query into a single function. At the moment this function contains multiple variable variables and function calls to convert the various semver formats when all you really need at this point is just the lowerSemverObj and upperSemverObj. Also your exclusivity logic doesn't currently work as it doesn't allow for half the semver query being inclusive.

Once you've got your `semverRangeStandardised` and checked it's not null you could capture all the logic up to this point of the mongo query in a single function, something like:

```
function parseSemverQuery(semverRangeStandardised: string): {
  upper: { major: string; minor: string; patch: string; inclusive: boolean }
  lower: { major: string; minor: string; patch: string; inclusive: boolean }
} {
  //parse semverRangeStandardised 
}
```
Then to reduce duplication in the inclusivity logic I would use if statements to set either `$gte` or `$gt` and `$lse` or `$ls` rather than repeat all the other bits of the mongo query in all the conditions. Eg:
```
if (semverQuery.upper.inclusive) {
  lowerComparator = '$gte' 
} else {
  lowerComparator = '$gt' 
}
if (semverQuery.lower.inclusive) {
  uppercomparator = '$lse' 
} else {
  uppercomparator = '$ls' 
}
```
Then you should be able to use this two variable in the relevant places in your single mongo query object Eg `'semver.major': { [lowerComparator]: semverQuery.lower.major }`

Reply:
Changes to semverRangeStandardised added, also added dynamic query stuff for inclusivity.